Decred Wallet/Features/Navigation Menu/NavigationMenuTabBarController.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  NavigationTabBarController.swift
+//  Decred Wallet
+//
+// Copyright (c) 2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+import UIKit
+
+class NavigationMenuTabBarController: UITabBarController {
+    
+    var isNewWallet: Bool = false
+    var customTabBar: CustomTabMenuView!
+    lazy var floatingButtons: NavMenuFloatingButtons = {
+        return NavMenuFloatingButtons()
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.loadTabBar()
+        if self.isNewWallet {
+            self.showOkAlert(message: LocalizedStrings.newWalletMsg)
+        }
+    }
+    
+    func loadTabBar() {
+        let tabItems: [MenuItem] = [.overview, .transactions, .wallets, .more]
+        self.setupCustomTabMenu(tabItems)
+        self.selectedIndex = 0
+    }
+    
+    // Create our custom menu bar and display it right where the tab bar should be.
+    func setupCustomTabMenu(_ menuItems: [MenuItem]) {
+        tabBar.isHidden = true
+        self.customTabBar = CustomTabMenuView(menuItems: menuItems, frame: tabBar.frame) // Draw and layout the tab navigation menu
+        
+        self.view.addSubview(customTabBar)
+        self.customTabBar.translatesAutoresizingMaskIntoConstraints = false // We are setting positioning constraints in the next line. best to ignore XCode generated constraints
+        self.customTabBar.clipsToBounds = true
+        
+        // Positioning constraints to place the nav menu right where the tab bar should be
+        NSLayoutConstraint.activate([
+            self.customTabBar.leadingAnchor.constraint(equalTo: tabBar.leadingAnchor),
```

Comment:
Since the `tabBar` is hidden before this point, wouldn't these constraints be inaccurate?

Reply:
it is only hidden, it is still in the view hierarchy and its positioning constraints are still valid.